src/kakarot/instructions/memory_operations.cairo

```diff
@@ -129,6 +139,12 @@ namespace MemoryOperations {
             return evm;
         }
 
+        if (offset.high != 0) {
+            let (revert_reason_len, revert_reason) = Errors.programCounterOutOfRange();
```

Comment:
should this error be: InvalidJumpDestinationError?
https://github.com/ethereum/execution-specs/blob/master/src/ethereum/shanghai/vm/exceptions.py#L69